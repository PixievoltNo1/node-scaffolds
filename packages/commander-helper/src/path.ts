import path from 'path'

/**
 * Calc absolute path of p under the workspace
 *
 * @param workspace
 * @param targetPath
 */
export function absoluteOfWorkspace(
  workspace: string,
  targetPath?: string | null,
): string {
  if (targetPath == null) return workspace
  const absoluteDir: string = path.resolve(workspace, targetPath)
  return absoluteDir
}

/**
 * Calc relative path to workspace
 *
 * @param workspace
 * @param targetDir
 */
export function relativeOfWorkspace(
  workspace: string,
  targetDir: string,
): string {
  const absoluteDir = absoluteOfWorkspace(workspace, targetDir)
  const relativeDir: string = path.relative(workspace, absoluteDir)
  return path.normalize(relativeDir)
}
