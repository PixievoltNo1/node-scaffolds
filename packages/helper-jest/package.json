{
  "name": "@guanghechen/helper-jest",
  "version": "4.7.3",
  "description": "A collection of utility functions for jest tests",
  "author": {
    "name": "guanghechen",
    "url": "https://github.com/guanghechen/"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/guanghechen/node-scaffolds/tree/release-4.x.x",
    "directory": "packages/helper-jest"
  },
  "homepage": "https://github.com/guanghechen/node-scaffolds/tree/release-4.x.x/packages/helper-jest#readme",
  "keywords": [],
  "type": "module",
  "main": "./lib/cjs/index.cjs",
  "module": "./lib/esm/index.mjs",
  "exports": {
    "import": "./lib/esm/index.mjs",
    "require": "./lib/cjs/index.cjs"
  },
  "types": "lib/types/index.d.ts",
  "source": "src/index.ts",
  "license": "MIT",
  "engines": {
    "node": ">= 16.0.0"
  },
  "files": [
    "lib/",
    "src/",
    "package.json",
    "CHANGELOG.md",
    "LICENSE",
    "README.md"
  ],
  "scripts": {
    "build": "rimraf lib/ && cross-env NODE_ENV=production rollup -c ../../rollup.config.mjs",
    "prepublishOnly": "yarn build",
    "test": "cross-env TS_NODE_FILES=true NODE_OPTIONS=--experimental-vm-modules jest --config ../../jest.config.mjs --rootDir ."
  },
  "dependencies": {
    "@guanghechen/helper-func": "^4.7.3",
    "@guanghechen/helper-is": "^4.7.3",
    "@jest/globals": "^29.5.0",
    "jest": "^29.5.0"
  },
  "devDependencies": {
    "cross-env": "^7.0.3",
    "rimraf": "^4.4.1",
    "rollup": "^3.20.2"
  }
}
