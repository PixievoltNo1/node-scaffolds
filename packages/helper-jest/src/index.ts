export * from './desensitizer/json'
export * from './desensitizer/types'
export * from './desensitizer/string'
export * from './mock/console'
export * from './mock/logger'
export * from './snapshot/file'
