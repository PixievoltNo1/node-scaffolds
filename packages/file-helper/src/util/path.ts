import path from 'path'

/**
 * Calc absolute path of p under the workspace
 *
 * @param workspace
 * @param targetPath
 */
export function absoluteOfWorkspace(workspace: string, targetPath?: string | null): string {
  if (targetPath == null) return workspace
  const filepath: string = path.resolve(workspace, targetPath)
  return path.normalize(filepath)
}

/**
 * Calc relative path to workspace
 *
 * @param workspace
 * @param targetPath
 */
export function relativeOfWorkspace(workspace: string, targetPath: string): string {
  const absoluteDir = absoluteOfWorkspace(workspace, targetPath)
  const filepath: string = path.relative(workspace, absoluteDir)
  return path.normalize(filepath)
}
