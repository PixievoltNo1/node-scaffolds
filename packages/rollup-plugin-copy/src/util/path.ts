import path from 'path'

/**
 * Resolve the source path.
 * @param workspace
 * @param srcPath
 * @returns
 */
export const resolvePath = (workspace: string, srcPath: string): string => {
  const filepath = path.resolve(workspace, srcPath)
  return filepath
}

/**
 * Get the path relative to workspace.
 * @param workspace
 * @param filepath
 * @returns
 */
export const relativePath = (workspace: string, filepath: string): string => {
  const srcPath: string = filepath.startsWith(workspace)
    ? filepath.slice(workspace.length).replace(/^[/\\]/, '')
    : filepath
  return srcPath
}
