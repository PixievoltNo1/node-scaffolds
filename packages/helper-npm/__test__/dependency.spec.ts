import type { IConsoleMock } from '@guanghechen/helper-jest'
import { createConsoleMock } from '@guanghechen/helper-jest'
import { desensitize, locateFixtures } from 'jest.helper'
import path from 'node:path'
import url from 'node:url'
import {
  collectAllDependencies,
  getDefaultDependencyFields,
  locateLatestPackageJson,
} from '../src/dependency'

describe('getDefaultDependencyFields', () => {
  test('basic', () => {
    expect(getDefaultDependencyFields()).toEqual([
      'dependencies',
      'optionalDependencies',
      'peerDependencies',
    ])
  })
})

describe('collectAllDependencies', () => {
  let logMock: IConsoleMock
  beforeEach(() => {
    logMock = createConsoleMock(['warn'])
  })
  afterEach(async () => {
    logMock.restore()
  })

  test('current repo', async () => {
    const __dirname = path.dirname(url.fileURLToPath(import.meta.url))
    const dependencies = await collectAllDependencies(path.join(__dirname, '../package.json'))
    expect(dependencies).toEqual(['@guanghechen/helper-path', 'import-meta-resolve'])

    expect(desensitize(logMock.getIndiscriminateAll())).toEqual([
      ["cannot find package.json for '@guanghechen/helper-path'"],
      ["cannot find package.json for 'import-meta-resolve'"],
    ])
  })

  test('nonexistent repo', async () => {
    const dependencies = await collectAllDependencies(
      locateFixtures('nonexistent/package.json'),
      undefined,
      ['rollup'],
      () => true,
    )
    expect(dependencies).toEqual(['rollup'])

    expect(desensitize(logMock.getIndiscriminateAll())).toEqual([
      [
        'no such file or directory: <$WORKSPACE$>/packages/helper-npm/__test__/fixtures/nonexistent/package.json',
      ],
      ["cannot find package.json for 'rollup'"],
    ])
  })

  test('normal repo', async () => {
    const dependencies = await collectAllDependencies(locateFixtures('normal-repo/package.json'), [
      'peerDependencies',
    ])
    expect(dependencies).toEqual(['@guanghechen/not-existed-repo', 'rollup'])

    expect(desensitize(logMock.getIndiscriminateAll())).toEqual([
      ["cannot find package.json for '@guanghechen/not-existed-repo'"],
      ["cannot find package.json for 'rollup'"],
    ])
  })
})

describe('locateLatestPackageJson', () => {
  test('basic', () => {
    const __dirname = path.dirname(url.fileURLToPath(import.meta.url))
    expect(locateLatestPackageJson(__dirname)).toBe(path.join(__dirname, '../package.json'))
  })
})
