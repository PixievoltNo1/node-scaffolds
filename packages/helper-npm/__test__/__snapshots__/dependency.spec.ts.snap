// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collectAllDependencies basic: current repo 1`] = `
[
  "@guanghechen/helper-path",
  "@guanghechen/invariant",
]
`;

exports[`collectAllDependencies basic: monorepo2 1`] = `
[
  "@babel/core",
  "@ampproject/remapping",
  "@jridgewell/gen-mapping",
  "@jridgewell/set-array",
  "@jridgewell/sourcemap-codec",
  "@jridgewell/trace-mapping",
  "@jridgewell/resolve-uri",
  "@babel/code-frame",
  "@babel/highlight",
  "@babel/helper-validator-identifier",
  "chalk",
  "ansi-styles",
  "color-convert",
  "color-name",
  "supports-color",
  "has-flag",
  "js-tokens",
  "@babel/generator",
  "@babel/types",
  "@babel/helper-string-parser",
  "to-fast-properties",
  "jsesc",
  "@babel/helper-compilation-targets",
  "@babel/compat-data",
  "@babel/helper-validator-option",
  "browserslist",
  "caniuse-lite",
  "electron-to-chromium",
  "node-releases",
  "update-browserslist-db",
  "escalade",
  "picocolors",
  "semver",
  "lru-cache",
  "yallist",
  "@babel/helper-module-transforms",
  "@babel/helper-environment-visitor",
  "@babel/helper-module-imports",
  "@babel/helper-simple-access",
  "@babel/helper-split-export-declaration",
  "@babel/template",
  "@babel/parser",
  "@babel/traverse",
  "@babel/helper-function-name",
  "@babel/helper-hoist-variables",
  "debug",
  "ms",
  "globals",
  "type-fest",
  "@babel/helpers",
  "convert-source-map",
  "safe-buffer",
  "gensync",
  "json5",
  "rollup",
  "fsevents",
]
`;

exports[`collectAllDependencies basic: normal-repo -- warnings 1`] = `
[
  "cannot find package.json for '@guanghechen/not-existed-repo'",
]
`;

exports[`collectAllDependencies basic: normal-repo 1`] = `
[
  "@guanghechen/not-existed-repo",
  "rollup",
]
`;
